JoinColony/colonyCDapp
# Cancelling non-staked expenditure as non-owner (using permissions)

## Description

This PR modifies the existing `cancelDraftExpenditure` saga to extends its functionality by new use cases:
- Cancelling locked expenditure
- Cancelling expenditure as non-owner (using permissions)

block-ingestor PR: https://github.com/JoinColony/block-ingestor/pull/183

## Testing

1. Create an expenditure and lock it, wait for the transaction to complete and refetch the expenditure using the testing UI.
2. Click on "Cancel expenditure" button.
3. The expenditure status should update to `CANCELLED`.
```gql
query ListExpenditures {
  listExpenditures(filter: {nativeId: {ge: 33}}) {
    items {
      nativeId
      status
    }
  }
}
```

Resolves #1997
